src/traffic_cop/traffic_cop.cpp

```diff
@@ -104,12 +104,13 @@ ResultType TrafficCop::CommitQueryHelper() {
   // If this exception is caused by a query in a transaction,
   // I will block following queries in that transaction until 'COMMIT' or
   // 'ROLLBACK' After receive 'COMMIT', see if it is rollback or really commit.
+
   if (curr_state.second != ResultType::ABORTED) {
     // txn committed
-    return txn_manager.CommitTransaction(txn);
+    return txn_manager.CommitTransaction(txn, GetOnCompleteCallback());
   } else {
     // otherwise, rollback
-    return txn_manager.AbortTransaction(txn);
+    return txn_manager.AbortTransaction(txn, GetOnCompleteCallback());
```

Comment:
I'm wondering if a callback is necessary for abort.

Reply:
Yes, we changed it do this.